lib/binding/kvdb_interface.c

```diff
@@ -1060,13 +1060,17 @@ hse_kvs_cursor_read(
     bool *                 eof)
 {
     merr_t err;
+    size_t vlen_dummy;
 
-    if (HSE_UNLIKELY(!cursor || !key || !klen || !val || !vlen || !eof || flags != 0))
+    if (HSE_UNLIKELY(!cursor || !key || !klen || !eof || flags != 0))
```

Comment:
Is klen really required? If user doesn't want that information, shouldn't we support it?

Reply:
Well, just having a pointer for the key without knowing its length doesn't seem meaningful.

Reply:
Keys and values are not necessarily NULL terminated.  So the length is always explicit.

Reply:
I understand, but that just seems like either a failure from the user perspective or I know the length of my keys beforehand. If HSE doesn't require the klen to be passed, then I don't understand why it is an error.

Reply:
Will fix this.

Reply:
Will fix this in a different PR.